rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/KubernetesApiImpl.scala

```diff
@@ -180,6 +184,54 @@ PUTs must contain resourceVersions. Response:
     } yield result
   }
 
+  /**
+   * Start a proxy in local minikube:
+   * - kubectl proxy --port=8080
+   * - replic_set_name=$(curl -s http://localhost:8080/api/v1/namespaces/akka-rollingupdate-demo-ns/pods | jq '.items[0].metadata.ownerReferences[0].name'| tr -d '"')
+   * - revision=$(curl -s http://localhost:8080/apis/apps/v1/namespaces/akka-rollingupdate-demo-ns/replicasets/"$replic_set_name" | jq '.metadata.annotations["deployment.kubernetes.io/revision"]'| tr -d '"')
+   *
+   * @return
+   */
+  override def readRevision(): Future[String] = {
+    val maxTries = 5
+    def loop(tries: Int = 0): Future[Option[String]] = {
+      val powOwnerRef = getPod().map { pod =>
+        for {
+          owner <- pod.metadata.ownerReferences.headOption
+        } yield {
+          owner
+        }
+      }
```

Comment:
isn't this the same as
```
getPod().map(_.metadata.ownerReferences.headOption)
```
?

Reply:
Can it have more than one owner reference, should we instead try to find the one with the ReplicaSet naming convention?

Reply:
Yes, the short version is better.

I have to read up under what circumstances a pod can have multiple ownerReferences.

Reply:
> should we instead try to find the one with the ReplicaSet naming convention?

Yes, I will filter the `ownerReferences` list to only look at items that are `"kind": "ReplicaSet"` and take the `headOption` of that filtered list (which should then only ever contain a single element after the filtering)